test/fuzz/transformation_wrap_vector_synonym_test.cpp

```diff
@@ -0,0 +1,401 @@
+         %89 = OpLoad %31 %35
+         %90 = OpFMul %31 %88 %89
+               OpStore %87 %90
+         %92 = OpLoad %31 %33
+         %93 = OpLoad %31 %35
+         %94 = OpFDiv %31 %92 %93
+               OpStore %91 %94
+               OpReturn
+               OpFunctionEnd
+  )";
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+
+  // Check context validity.
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Vec Type Id |   Vector Type  |  Element Type id |   Element Type  |
+  // ------------+----------------+------------------+-----------------+
+  //     12      |      vec2      |         6        |      int32      |
+  //     24      |      vec3      |        18        |     uint32      |
+  //     37      |      vec4      |        31        |      float      |
+
+  // Instruction Id | Opcode  | Type Id | constant id 1 | constant id 2 |
+  // ---------------+---------+---------+---------------+---------------+
+  //       50       | OpIAdd  |    6    |      48       |      49       |
+  //       54       | OpISub  |    6    |      52       |      53       |
+  //       58       | OpIMul  |    6    |      56       |      56       |
+  //       62       | OpSDiv  |    6    |      60       |      61       |
+  //       66       | OpIAdd  |    18   |      64       |      65       |
+  //       70       | OpISub  |    18   |      68       |      69       |
+  //       74       | OpIMul  |    18   |      72       |      73       |
+  //       78       | OpUDiv  |    18   |      76       |      77       |
+  //       82       | OpFAdd  |    31   |      80       |      81       |
+  //       86       | OpFSub  |    31   |      84       |      85       |
+  //       90       | OpFMul  |    31   |      88       |      89       |
+  //       94       | OpFDiv  |    31   |      92       |      93       |
+
+  TransformationCompositeConstruct add_int_vec1(
+      12, {48, 48}, MakeInstructionDescriptor(50, SpvOpIAdd, 0), 100);
+  ASSERT_TRUE(add_int_vec1.IsApplicable(context.get(), transformation_context));
+
+  TransformationCompositeConstruct add_int_vec2(
+      12, {49, 49}, MakeInstructionDescriptor(50, SpvOpIAdd, 0), 101);
+  ASSERT_TRUE(add_int_vec2.IsApplicable(context.get(), transformation_context));
+  // Insert vec2 of id 100 with the first value of OpIAdd instruction with
+  // id 50.
+  ApplyAndCheckFreshIds(add_int_vec1, context.get(), &transformation_context);
+  // Insert vec2 of id 101 with the second value of OpIAdd instruction with
+  // id 50.
+  ApplyAndCheckFreshIds(add_int_vec2, context.get(), &transformation_context);
```

Comment:
Can you remove this, and instead adapt the shader so that it is in the form it would be in if these transformations had been applied?

We try to make tests as independent and self contained as possible, so that in the tests for one transformation we don't apply other kinds of transformation.

Reply:
Done.